packages/kit-headless/src/components/modal/modal-trigger.tsx

```diff
@@ -0,0 +1,16 @@
+import { PropsOf, Slot, component$, useContext, $ } from '@builder.io/qwik';
+import { modalContextId } from './modal-context';
+
+export const ModalTrigger = component$((props: PropsOf<'button'>) => {
+  const context = useContext(modalContextId);
+
+  const handleClick$ = $(() => {
+    context.showSig.value = !context.showSig.value;
+  });
+
+  return (
+    <button onClick$={[handleClick$, props.onClick$]} {...props}>
+      <Slot />
+    </button>
```

Comment:
Shouldn't the ModalTrigger button also have some aria attributes on it?

Like in Radix:

```tsx
const DialogTrigger = React.forwardRef<DialogTriggerElement, DialogTriggerProps>(
  (props: ScopedProps<DialogTriggerProps>, forwardedRef) => {
    const { __scopeDialog, ...triggerProps } = props;
    const context = useDialogContext(TRIGGER_NAME, __scopeDialog);
    const composedTriggerRef = useComposedRefs(forwardedRef, context.triggerRef);
    return (
      <Primitive.button
        type="button"
        aria-haspopup="dialog"
        aria-expanded={context.open}
        aria-controls={context.contentId}
        data-state={getState(context.open)}
        {...triggerProps}
        ref={composedTriggerRef}
        onClick={composeEventHandlers(props.onClick, context.onOpenToggle)}
      />
    );
  }
);
```

Also I feel like if we add a ModalTrigger we'll need to add an `asChild` prop to it. This way users can directly put their `<Button>` component in the ModalTrigger.

Reply:
Yes, the following aria attributes should be added to the modal trigger, thanks for pointing that out.

As for the asChild pattern, I don't see the large benefit. The component needs to know its direct child, which goes against the philosophy of both slots and Qwik's out of order rendering.

To be consistent, then half the library would need an `asChild` prop, which lengthens the markup. I'm not sure having an inline component everywhere would be the best idea either.

I don't see the article you've linked. In either case, a trigger is more the most part a button. There are two reasons for the switch:

1. A better developer experience and a11y as a default.
2. Invokers. **https://open-ui.org/components/invokers.explainer/**

Reply:
Lol sorry, forgot to link it: https://medium.com/@bryanmylee/aschild-in-react-svelte-vue-and-solid-for-render-delegation-645c73650ced